### DISCUSSION TITLE
Is it possible for the events to support multiple signatures? like in overloading

### DISCUSSION THREAD

**Discussion by RolandMiculas** [miguelgrinberg/python-socketio#764] [Q&A] [answered]
I've tried multiple simple implementations for receiving event such as multipledispatch, using *args and **kargs, but none of them seems to work, could you provide an example for such a thing?

**Comment by miguelgrinberg** [comment, marked as answer]
What have you tried? I haven't done this, but I think it should work with the `@sio.on()` decorator, which should be stackable.

**Reply by RolandMiculas** [reply]
```
@sio.on('stream', namespace=namespace)
def stream(**kwargs):
```
 
```
@sio.on('stream', namespace=namespace)
def stream(*args, ):
```

```
@sio.on('stream', namespace=namespace)
def stream(sid, message):

@sio.on('stream', namespace=namespace)
def stream2(sid):
```

```
 from multipledispatch import dispatch

@dispatch(type(""), type({}))
@sio.on("stream", namespace=namespace)
def stream(sid, message):

@dispatch(type(""))
@sio.on("stream", namespace=namespace)
def stream(sid):
```

and on the client side i'm just trying to do 

```
sq = sio.call('stream', namespace=namespace)
s = sio.call('stream', data, namespace=namespace)
```

note, im trying to send a stream event with and without data

**Reply by miguelgrinberg** [reply]
Sorry, I think I misunderstood. I assumed you wanted several event signatures to be mapped to the same handler function. This is what I was thinking you wanted to do:

```python
@sio.on('stream', namespace=namespace)
@sio.on('stream1', namespace=namespace)
@sio.on('stream2', namespace=namespace)
def all_the_streams_come_here(*args):
    pass
```

But it seems you want a single event to fire multiple handler functions, which is not supported. You can implement this manually by calling all the functions you need from a single handler function.
